### PR TITLE
Alert and task to refresh saml provider metadata

### PR DESCRIPTION
**Pull request by shaidar**
#### What are the relevant tickets?
- [Issue#1133](https://github.com/mitodl/salt-ops/issues/1133)
- [Issue#1132](https://github.com/mitodl/salt-ops/issues/1132)

### CODE REVIEW

File: pillar/edx/init.sls [hunk 1]
```diff
@@ -47,3 +47,10 @@ schedule:
     args:
       - etl.mitx
   {% endif %}
+  {% if 'mitx-' in environment %}
```
Comment (blarghmatey) [inline pillar/edx/init.sls]: To be more explicit about this we can filter on `business_unit == residential`

File: pillar/elastalert.sls [hunk 2]
```diff
@@ -124,6 +124,36 @@ elastic_stack:
...
+        settings:
+          name: Authentication with MIT Kerberos is currently unavailable
+          description: >-
+            Send a message anytime an error message containing No SAMLProviderData
+            found is encountered in the mitx production logs.
+          opsgenie_key: {{ opsgenie_key }}
+          opsgenie_priority: P1
+          opsgenie_alias: mitx_saml_error
+          type: frequency
+          index: logstash-*
```
Comment (blarghmatey) [inline pillar/elastalert.sls]: Should we segment this to logstash-mitx-production-*?

File: pillar/elastalert.sls [hunk 3]
```diff
@@ -124,6 +124,36 @@ elastic_stack:
...
+          num_events: 5
```
Comment (blarghmatey) [inline pillar/elastalert.sls]: Should this be triggered on 1 event?

File: pillar/edx/init.sls (hunk 1 above)
Comment (shaidar) [reply pillar/edx/init.sls]: Sure, but feels as if we're more consistent with the environment grain than the business_unit, but I'll set it as much

File: pillar/elastalert.sls (hunk 2 above)
Comment (shaidar) [reply pillar/elastalert.sls]: We want the alert for QA too though, no?

Comment (blarghmatey) [reply pillar/elastalert.sls]: Well, we only want to get paged for production

Comment (blarghmatey) [reply pillar/elastalert.sls]: Also, the environment name is already being set to production at the bottom of this stanza

**Comment by blarghmatey** [comment]
:+1:

**Pull request merged**